src/index.ts

```diff
@@ -26,13 +26,19 @@ function loader(this: interfaces.Webpack, contents: string) {
         return;
     }
 
+    // Load files generated by previous loaders in the same chain
+    const generatedFiles = getGeneratedFiles(this);
+    for (let filePath of Object.keys(generatedFiles)) {
+        updateFileInCache(filePath, generatedFiles[filePath], instance);
+    }
+
     const rawFilePath = path.normalize(this.resourcePath);
     const filePath = utils.appendTsSuffixIfMatch(options.appendTsSuffixTo, rawFilePath);
     const fileVersion = updateFileInCache(filePath, contents, instance);
 
     const { outputText, sourceMapText } = options.transpileOnly
         ? getTranspilationEmit(filePath, contents, instance, this)
```

Comment:
Is it significant that it's not passed down to `getTranspilationEmit` but is to `getEmit`?